atomic_reactor/inner.py

```diff
@@ -493,9 +497,13 @@ def build_docker_image(self):
         finally:
             # We need to make sure all exit plugins are executed
             signal.signal(signal.SIGTERM, lambda *args: None)
-            exit_runner = ExitPluginsRunner(self.builder.tasker, self,
-                                            self.exit_plugins_conf,
-                                            plugin_files=self.plugin_files)
+            try:
+                exit_runner
+            except NameError:
```

Comment:
I don't understand what's going on here.  Try the exit_runner and if it generates a key_error, assign the exact same variable and run it again.  How is this an improvement over the old way?

Reply:
We now create instances of each type of runner in the beginning of the build process. The runners may do some work upon initialization, which can lead to exceptions. The exit runner is also initialized with all the other runners. If a runner fail to initialize, the code will proceed to run the exception section and then move to the `finally` statement. If the exception occurs in any runner initialization other than the exit_runner, it needs to be initialized here.

Reply:
Could be use `exit_runner = None` instead? I don't really like catching `NameError`